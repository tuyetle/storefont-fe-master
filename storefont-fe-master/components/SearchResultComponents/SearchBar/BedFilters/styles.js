import { css } from 'glamor';
import variables from '~/styles/variables';

export default {
  bedFilters: css({
    position: 'relative',
    fontSize: '0.8em',
    width: '25em',
    top: '0em',
    padding: '0 0 1.8em 0',
    background: 'white',
    textAlign: 'left',
    '@media (min-width: 768px)': {
      position: 'relative',
      fontSize: '0.8em',
      width: '25em',
      top: '0em',
      padding: '1em',
      background: 'white',
      boxShadow: variables.$boxShadowPrice,
      textAlign: 'auto',
    },
  }),
  item: css({
    margin: '0 2em 0 0',
    '& input[type=checkbox]': {
      display: 'inline-block',
      verticalAlign: 'middle',
    },
    '@media (min-width: 768px)': {
      margin: '0 1em',
      '& input[type=checkbox]:checked:before': {
        top: '-0.8em',
      },
    },
  }),
  textInfo: css({
    paddingLeft: '0.5em',
  }),
  item_input: css({
    marginLeft: '0',
  }),
  item_span: css({ paddingLeft: '0.5em', position: 'absolute' }),
};
