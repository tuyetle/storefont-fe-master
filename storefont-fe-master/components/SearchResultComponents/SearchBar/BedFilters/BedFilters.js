import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { isEmpty } from 'lodash';
import Checkbox from '~/components/common/Checkbox/Checkbox';
import styles from './styles';

class BedFilters extends Component {
  constructor(props) {
    super(props);
    let arr = [];
    const {
      fieldApi: { value },
    } = props;
    if (value) {
      arr = value;
    } else if (props.defaultValue) {
      arr = props.defaultValue;
    }

    this.state = { value: arr };
  }

  onChangeHandle = value => (event) => {
    const arr = [...this.state.value];
    if (event.target.checked) {
      arr.push(value);
    } else {
      arr.splice(arr.indexOf(value), 1);
    }

    const {
      fieldApi: { setValue, setTouched },
    } = this.props;
    setTouched(true);
    setValue(arr);
    this.setState({ value: arr });
  };

  isContainInDefaultValue = (value) => {
    const arr = this.state.value;
    if (!isEmpty(arr) && arr.includes(value)) {
      return true;
    }

    return false;
  };

  render() {
    return (
      <section className={`${styles.bedFilters}`}>
        <div>
          {
            this.props.bedOptionsConfig.map(({ label, value }) => (
              <label className={`${styles.item}`} key={label}>
                <Checkbox
                  value={value}
                  onChange={this.onChangeHandle(value)}
                  defaultChecked={this.isContainInDefaultValue(value)}
                />
                <span className={`${styles.textInfo}`} >{label}</span>
              </label>
            ))
          }
        </div>
      </section>
    );
  }
}

BedFilters.propTypes = {
  fieldApi: PropTypes.object.isRequired,
  defaultValue: PropTypes.array,
  bedOptionsConfig: PropTypes.arrayOf(PropTypes.shape({
    value: PropTypes.number.isRequired,
    label: PropTypes.string.isRequired,
  }).isRequired).isRequired,
};

BedFilters.defaultProps = {
  defaultValue: [],
};

export default BedFilters;
