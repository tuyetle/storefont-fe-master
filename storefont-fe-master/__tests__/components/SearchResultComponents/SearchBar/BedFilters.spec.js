import React from 'react';
import renderer from 'react-test-renderer';
import BedFilters from '~/components/SearchResultComponents/SearchBar/BedFilters/BedFilters';

describe('(Component) BedFilters', () => {
  let component;
  let props;
  beforeEach(() => {
    props = {
      fieldApi: {
        setValue: () => { },
        setTouched: () => { },
      },
      bedOptionsConfig: [{
        value: 1,
        label: '1',
      },
      {
        value: 2,
        label: '2',
      }],
    };

    component = (<BedFilters {...props} />);
  });

  it('should render conrectly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

