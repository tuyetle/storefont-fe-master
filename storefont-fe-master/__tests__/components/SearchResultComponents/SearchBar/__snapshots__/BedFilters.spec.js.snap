// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) BedFilters should render conrectly 1`] = `
<section
  className="css-31twqw"
>
  <div>
    <label
      className="css-u2zv12"
    >
      <input
        className="css-1k7inrl"
        defaultChecked={false}
        onChange={[Function]}
        type="checkbox"
        value={1}
      />
      <span
        className="css-ww42u9"
      >
        1
      </span>
    </label>
    <label
      className="css-u2zv12"
    >
      <input
        className="css-1k7inrl"
        defaultChecked={false}
        onChange={[Function]}
        type="checkbox"
        value={2}
      />
      <span
        className="css-ww42u9"
      >
        2
      </span>
    </label>
  </div>
</section>
`;
